### PR TITLE
Dion tag frontend and functionalities

### PR DESCRIPTION
**Pull request by dionyichia**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
sprint as described in linear

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
overall view:
image

Flitered view:
image

post tag association deletion:
image

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

run the app, got to template-link page.tsx, currently defaults to a user-group-id ive set, press on tags in sidebar to filter, press on delete crosss to delete, check db table template tag to see changes, press new tag to add new tag, check tag table for new row with 10 in number column, insert new template tag association to test if tag actually appears 

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- styling issues, now display is just  justify-content: space-between; for the template row, tags becomes misaligned if template name is longer or shorter
- functionality to edit the tag name in the sidebar

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @me-liu

### CODE REVIEW

File: components/tag/TagCreator.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          key={tag.tag_id}
+          $isSelected={selectedTagId === tag.tag_id}
+          onClick={() => onTagClick(tag.tag_id)}
+        >
+          <TagComponent
+            color={tag.color as ColorKey}
+            name={tag.name}
+            tag_id={tag.tag_id}
+            sidebar={true}
+            onRename={handleRename}
```
Comment (me-liu) [inline components/tag/TagCreator.tsx]: Can you add a cross for deletion in the sidebar next to each tag? This would delete the tag and also all connections in the tag-template table

**Review by me-liu — COMMENTED** [review]
yay functionality looks so solid! I edited some of the styles and added some comments for some changes

File: components/tag/TagCreator.tsx (hunk 1 above)
Comment (dionyichia) [reply components/tag/TagCreator.tsx]: alrighty done, when side bar x is clicked, it will delete all connections and update all rows

before
image

after
image

File: app/templates/template-list/page.tsx [hunk 2]
```diff
@@ -46,20 +68,46 @@ const SearchBar: React.FC = () => {
...
+
+      setTemplates(templatesWithTags);
+      setFilteredTemplates(templatesWithTags);
     };
     loadTemplates();
-  }, [user_group_id]);
+  }, [user_group_id, tagVersion]);
 
   // Filter based on search.
   useEffect(() => {
```
[the remark this answers is not in the document]
Comment (me-liu) [reply app/templates/template-list/page.tsx]: ok cool no worries i think the library is overkill hahah it works good

**Review by me-liu — APPROVED** [review]
